compiler/test/dotc/tests.scala

```diff
@@ -245,7 +245,7 @@ class tests extends CompilerTest {
   }
 
 
-  @Test def compileStdLib = compileList("compileStdLib", stdlibFiles, "-migration" :: "-Yno-inline" :: scala2mode)
+  @Test def compileStdLib = compileList("compileStdLib", stdlibFiles.map(_.replace("../scala-scala/src/library/", "")), "-migration" :: "-Yno-inline" :: scala2mode, Directory("../scala-scala/src/library/"))
   @Test def compileMixed = compileLine(
```

Comment:
I've a minor question here: for other calls of `compileList`, we ignore the option `sourceDir`. It would be nice to add a comment for `compileList` to document the condition when this option can be ignored.